src/jbi/bugzilla_objects.py

```diff
@@ -0,0 +1,244 @@
+        converted_list: List = []
+        for whiteboard in self.get_whiteboard_as_list():
+            converted_tag = self.convert_wb_to_tag(whiteboard=whiteboard)
+            if converted_tag not in [None, "", " "]:
+                converted_list.append(converted_tag)
+
+        return converted_list
+
+    @staticmethod
+    def convert_wb_to_tag(whiteboard):
+        """Extract tag from whiteboard label"""
+        _exists = whiteboard not in (" ", "")
+        if not _exists:
+            return ""
+        return whiteboard.split(sep="-", maxsplit=1)[0].lower()
+
+    def get_jira_issue_dict(self):
+        """Extract bug info as jira issue dictionary"""
+        type_map: dict = {"enhancement": "Task", "task": "Task", "defect": "Bug"}
+        return {
+            "summary": self.summary,
+            "labels": self.get_jbi_labels(),
+            "issuetype": {"name": type_map.get(self.type, "Task")},
+        }
+
+    def extract_from_see_also(self):
+        """Extract Jira Issue Key from see_also if jira url present"""
+        if not self.see_also and len(self.see_also) > 0:
+            return None
+
+        for url in self.see_also:  # pylint: disable=not-an-iterable
+            try:
+                parsed_url: ParseResult = urlparse(url=url)
+                expected_hosts = ["jira", "atlassian"]
+
+                if any(  # pylint: disable=use-a-generator
+                    [part in expected_hosts for part in parsed_url.hostname.split(".")]
+                ):
+                    parsed_jira_key = parsed_url.path.split("/")[-1]
+                    return parsed_jira_key
+            except Exception:  # pylint: disable=broad-except
+                # Try parsing all see_also fields; skip errors.
+                continue
+        return None
+
+
+class BugzillaWebhookRequest(BaseModel):
+    """Bugzilla Webhook Request Object"""
+
+    webhook_id: int
+    webhook_name: str
+    event: BugzillaWebhookEvent
+    bug: Optional[BugzillaBug]
+
+    def map_as_jira_comment(self):
+        """Extract comment from Webhook Event"""
+        comment: BugzillaWebhookComment = self.bug.comment
+        commenter: BugzillaWebhookUser = self.event.user
+        comment_body: str = comment.body
+        body = f"*({commenter.login})* commented: \n{{quote}}{comment_body}{{quote}}"
+        return body
+
+    def map_as_jira_description(self):
+        """Extract description as comment from Webhook Event"""
+        comment: BugzillaWebhookComment = self.bug.comment
+        comment_body: str = comment.body
+        body = f"*(description)*: \n{{quote}}{comment_body}{{quote}}"
+        return body
+
+    def map_as_tuple_of_field_dict_and_comments(
+        self,
+        status_log_enabled: bool = True,
+        assignee_log_enabled: bool = True,
+    ) -> Tuple[Dict, List]:
+        """Extract update dict and comment list from Webhook Event"""
+
+        comments: List = []
+        if not self.bug:
+            return {}, []
```

Comment:
Don't we want to raise a ValueError explicit here? It feels dangerous to have this silent implicit conversion to empty objects